Sources/StreamChatUI/ChatMessageList/MessageComposer/ChatMessageComposerMentionCellView_Tests.swift

```diff
@@ -0,0 +1,236 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import StreamChatTestTools
+@testable import StreamChatUI
+import XCTest
+
+class ChatMessageComposerMentionCellView_Tests: XCTestCase {
+    /// Default reference width for the cell on current tested device
+    private static var defaultCellWidth = UIScreen.main.bounds.width - 16
+
+    override func setUp() {
+        super.setUp()
+    }
+
+    func test_emptyAppearance() {
+        let view = ChatMessageComposerMentionCellView().withoutAutoresizingMaskConstraints
+        view.widthAnchor.constraint(equalToConstant: Self.defaultCellWidth).isActive = true
+
+        AssertSnapshot(view, variants: [.small])
+    }
+
+    func test_defaultAppearance_withUserOnline() {
+        let view = ChatMessageComposerMentionCellView().withoutAutoresizingMaskConstraints
+        view.widthAnchor.constraint(equalToConstant: Self.defaultCellWidth).isActive = true
+        view.content = ("Mr Vader", "darkforce37", TestImages.vader.url, true)
+
+        AssertSnapshot(view, variants: [.small])
+    }
+
+    func test_defaultAppearance_withUserOffline() {
+        let view = ChatMessageComposerMentionCellView().withoutAutoresizingMaskConstraints
+        view.widthAnchor.constraint(equalToConstant: Self.defaultCellWidth).isActive = true
+        view.content = ("Mr Vader", "darkforce37", TestImages.vader.url, false)
+
+        AssertSnapshot(view, variants: [.small])
+    }
+
+    func test_appearanceCustomization_usingUIConfig_withUserOnline() {
+        class RectIndicator: UIView {
+            override func didMoveToSuperview() {
+                super.didMoveToSuperview()
+                backgroundColor = .green
+                widthAnchor.constraint(equalTo: heightAnchor, multiplier: 1).isActive = true
+            }
+        }
+
+        class CustomAvatarView: ChatMessageComposerMentionAvatarView {
+            override func didMoveToSuperview() {
+                super.didMoveToSuperview()
+                backgroundColor = .green
+                widthAnchor.constraint(equalTo: heightAnchor, multiplier: 1).isActive = true
+            }
```

Comment:
if only we had a dedicated set of methods you should override to achieve certain customization ... oh wait!

Reply:
Thanks but resolved in Slack channel :trollface: